### ISSUE TITLE
Using deprecated function in documentation (reducerComponentWithRetainedProps)

### ISSUE THREAD

**Issue by simeg** [reasonml/reason-react#248]
Looking through the documentation I found this [example on on `willReceiveProps`](https://reasonml.github.io/reason-react/docs/en/lifecycles.html#willreceiveprops) and in the example the function `reducerComponentWithRetainedProps` is called to create the `component` variable.

When I try to use this in my code the compiler tells me `The retained props feature is deprecated`. This had me confused for a while until I found out that [it is actually deprecated](https://github.com/reasonml/reason-react/blob/master/HISTORY.md#024) and has been for almost 10 months. Well technically the list of changes doesn't even specify that `reducerComponentWithRetainedProps` is deprecated, only `statefulComponentWithRetainedProps`.

I'm submitting an issue instead of a PR because I don't know how the example would look like today (will dig into migration steps now). But I think this should be updated since it's been broken for quite some time now.

**Comment by sgny** [comment]
There's something amiss, you should not be getting that warning. Yet.

I see that retainedProps are marked to be replaced in the future with getDerivedStateFromProps, but that's in the medium/long term (see FUTURE.md).

I'm using both statelessComponentWithRetainedProps and reducerComponentWithRetainedProps in my app and I don't get the deprecation warning even with a clean build. You could also check the source in this repo and you'll find that neither function is tagged as deprecated.

In short, I don't see why you should be getting that warning. Maybe try asking on discord or another channel if you haven't already done so.

**Comment by simeg** [comment]
Hm, that's weird. I'm not working on my reason-react thing anymore but when I'll pick it up again I'm sure to investigate this more. Thanks.
